### ISSUE TITLE
Exploding Liquid fails with Verlet Lists Pairwise

### ISSUE THREAD

**Issue by SamNewcome** [AutoPas/AutoPas#975]
**Describe the bug**
Verlet Lists Pairwise breaks in combination with `pause-simulation-during-tuning`. In particular, when this seems to fail in combination with SoA during the first time SoA is trailed in the first tuning phase. See attached yaml file (as txt file).

**To Reproduce**
1. Run the attached yaml file
2. Set `pause-simulation-during-tuning` to false and rerun to see if there are no problems with this.

This doesn't appear entirely reproducible. With HSUper, it fails much later in the simulation. This setup is a little different: 2 node x 3 ranks per node x 24 CPUs per rank.

**Build environment (please complete the following information):**
* OS: Linux
* Compiler: GCC 11.4 (and 13.2)
* Build System: (e.g. Make, ninja) Make

**Additional context**
Yaml file:
[explodingLiquid2.txt](https://github.com/user-attachments/files/17845613/explodingLiquid2.txt)

**Comment by ManuelLerchner** [comment]
This issue caught my attention, because I had a similar problem in #995.

I think this is caused by not calling `_autoPasContainer->updateContainer();` whenever the simulation is paused, which means that the neighborlists are never rebuilt. This can cause crashes when the container type changes during tuning. If this is the actual cause I am surprised that this has not been caught earlier, it seems like a problem that should occur all the time...

Moving the `updateContainer()` function (and possible more?) outside the if-clause could help.

```diff
--- a/examples/md-flexible/src/Simulation.cpp
+++ b/examples/md-flexible/src/Simulation.cpp
@@ -221,16 +221,16 @@ void Simulation::run() {
     }

     _timers.computationalLoad.start();
+
+    _timers.updateContainer.start();
+    auto emigrants = _autoPasContainer->updateContainer();
+    _timers.updateContainer.stop();
+
     if (_configuration.deltaT.value != 0 and not _simulationIsPaused) {
       updatePositionsAndResetForces();
 #if MD_FLEXIBLE_MODE == MULTISITE
       updateQuaternions();
 #endif
-
-      _timers.updateContainer.start();
-      auto emigrants = _autoPasContainer->updateContainer();
-      _timers.updateContainer.stop();
-
       const auto computationalLoad = static_cast<double>(_timers.com
```

I havent fully completed the simulation and verified this, but at least it doesn't crash instantly anymore when running it locally.
Maybe this is worth a closer look.

**Comment by SamNewcome** [comment]
> Moving the `updateContainer()` function (and possible more?) outside the if-clause could help.

Yes, I implemented this as a quick hotfix and this seems to fix the problem
